src/NUnitEngine/nunit.engine.core/Drivers/NUnitFrameworkApi2018.cs

```diff
@@ -0,0 +1,293 @@
+#else
+    public class NUnitFrameworkApi2018 : NUnitFrameworkApi
+#endif
+    {
+        static readonly ILogger log = InternalTrace.GetLogger(nameof(NUnitFrameworkApi2018));
+
+        const string LOAD_MESSAGE = "Method called without calling Load first. Possible error in runner.";
+        const string INVALID_FRAMEWORK_MESSAGE = "Running tests against this version of the framework using this driver is not supported. Please update NUnit.Framework to the latest version.";
+        const string FAILED_TO_LOAD_ASSEMBLY = "Failed to load assembly ";
+        const string FAILED_TO_LOAD_NUNIT = "Failed to load the NUnit Framework in the test assembly";
+
+        const string CONTROLLER_TYPE = "NUnit.Framework.Api.FrameworkController";
+
+        private string _driverId;
+
+        AssemblyName _nunitRef;
+        string? _testAssemblyPath;
+
+        object? _frameworkController;
+        Type? _frameworkControllerType;
+
+
+        public NUnitFrameworkApi2018(string driverId, AssemblyName nunitRef)
+        {
+            Guard.ArgumentNotNull(driverId, nameof(driverId));
+            Guard.ArgumentNotNull(nunitRef, nameof(nunitRef));
+
+            _driverId = driverId;
+#if NETFRAMEWORK
+            //if (RemotingServices.IsTransparentProxy(driver))
+            //{
+            //    // We need to replace nunitRef with a reference valid in the current domain.
+            //    var frameworkRef = AssemblyNameReference.Parse(nunitRef.FullName);
+            //    var frameworkDef = new DefaultAssemblyResolver().Resolve(frameworkRef);
+            //    frameworkDef.MainModule.
+            //}
+            //if (_nunitRef == null)
+                _nunitRef = nunitRef;
+#else
+            _nunitRef = nunitRef;
+#endif
+        }
+
+#if NETCOREAPP
+        TestAssemblyLoadContext? _assemblyLoadContext;
+        Assembly? _testAssembly;
+        Assembly? _frameworkAssembly;
+#endif
+
+        public string Load(string testAssemblyPath, IDictionary<string, object> settings)
+        {
+            Guard.ArgumentNotNull(testAssemblyPath, nameof(testAssemblyPath));
+            Guard.ArgumentNotNull(settings, nameof(settings));
+            Guard.ArgumentValid(File.Exists(testAssemblyPath), "Framework driver called with a file name that doesn't exist.", "testAssemblyPath");
+            log.Info($"Loading {testAssemblyPath} - see separate log file");
+
+            _testAssemblyPath = Path.GetFullPath(testAssemblyPath);
+            var idPrefix = string.IsNullOrEmpty(_driverId) ? "" : _driverId + "-";
+
+#if NETFRAMEWORK
+            // Normally, the caller should check for an invalid requested runtime, but we make sure here
+            var requestedRuntime = settings.ContainsKey(EnginePackageSettings.RequestedRuntimeFramework)
+                ? settings[EnginePackageSettings.RequestedRuntimeFramework] : null;
+
+            try
+            {
+                _frameworkController = AppDomain.CurrentDomain.CreateInstanceAndUnwrap(
```

Comment:
As the Api itself is loaded in the _testdomain_, is there now any reason to now not to use the same code as for the .NET Core version?

Reply:
Ok, I see that the .NET Core version uses `AssemblyLoadContext` and .NET Framework would use `AppDomain`.
Note sure if either of them here have any effect.
It is used here to load the `FrameworkController`, but that controller directly calls `Assembly.Load` itself without doing any `AssemblyLoadContext` or `AppDomain` distinction.

If there is an effect, it could be abstracted in a `TestAssemblyLoader` class to hide the distinctions between the .NET Core and Framework.

Reply:
I think abstracting it is a good idea, but I'd rather get it working with separate code first. "To remove duplication, you must first create it."